src/components/map/Map.tsx

```diff
@@ -104,6 +104,18 @@ class LeafletMap extends React.Component<IMapProps, IMapState> {
         );
     }
 
+    private getFirstNodes() {
```

Comment:
We could simplify this since this function is mostly getting the visible route paths, but we already have this information in the visibleRoutePaths variable. So, what if we instead would pass these visibleRoutePaths to this getFirstNodes function, and it would just map these visibleRoutePaths by returning their start locations. Something like this:

    private getFirstNodes(visibleRoutePaths: IRoutePath[]) {
        return visibleRoutePaths.map(rPaths => rPaths.geoJson.coordinates[0]);
    }